CMakeLists.txt

```diff
@@ -258,10 +258,12 @@ else()
 endif()
 if(WITH_CODE_COVERAGE)
     if(CMAKE_C_COMPILER_ID MATCHES "Clang")
-        set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -O0 -g -coverage")
+        set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -coverage")
```

Comment:
Are you sure you want to remove `-O0`? Perhaps it will be better if it is `-O1`. If it is `O3` then maybe certain lines will be optimized away and code coverage % will decrease.

Btw, I think the sanitizers need `-O1`.

Reply:
The problem with using -O0 with coverage is that our CI then never even tests compiler optimizations, and that could be bad in itself. It'd be better if some of the CI runs are -O0 or -O1 instead of all of them.

Reply:
Modified a couple runs to compile with debug symbols, and one with -O1